DylibBinder/DBGenericParameter.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections.Generic;
+using Dynamo.SwiftLang;
+using SwiftReflector;
+using SwiftReflector.Inventory;
+
+namespace DylibBinder {
+	public class DBGenericParameter {
+		public DBGenericParameter (int depth, int index)
+		{
+			Depth = depth;
+			Index = index;
+			Name = SLGenericReferenceType.DefaultNamer (depth, index);
+		}
+
+		public int Depth { get; }
+		public int Index { get; }
+		public string Name { get; }
+	}
+
+	public class DBGenericParameters {
+
+		public DBGenericParameters (SwiftBaseFunctionType signature) {
+			AssignGenericParameters (signature);
+		}
+
+		public DBGenericParameters (SwiftType swiftType)
+		{
+			FilterGenericParameterSource (swiftType);
+		}
+
+		public DBGenericParameters (DBTypeDeclaration typeDeclaration)
+		{
+			ParseTopLevelGenerics (typeDeclaration.Funcs, typeDeclaration.Properties);
+		}
```

Comment:
This one here, does look like a nice candidate for a static method, and therefor a extension, right?

Reply:
Could you explain why these DBGenericParameter methods should be static?
These methods call different entry points in the methods at the end of the file, but the methods
you mentioned as being good candidates for extension methods are different types so wouldn't I need
to copy code into these extension classes?

Reply:
It is taking just the information of the DBTypeDeclaration, so we can provide a method that those the parsetoplevelgenerics and returns them. The way you implemented it makes the code very hard to test, which is going to be a problem once we start adding unittests. Because we are going to add tests.